### ISSUE TITLE
unexpected results when defining or outputting negative floating point number

### ISSUE THREAD

**Issue by shyoshyo** [LingDong-/wenyan-lang#32]
The following wenyan code:
```
吾有一數。曰負一又二分三釐四毫五絲六忽七微，書之。
```
is compiled to the following JavaScript equivalent code:
```JavaScript
var _ans138 = 1.234567;
console.log(_ans138);
```

However, it should be:
```JavaScript
var _ans138 = -1.234567;
console.log(_ans138);
```

---

The following wenyan code:
```
乘一又二分三釐四毫五絲六忽七微以負一。名之曰「甲」。
吾有一數。曰 「甲」。書之。
```
produced the following output by compiled JavaScript code:
```
負undefined
```
But the output should be:
```
負一又二分三釐四毫五絲六忽七微
```

**Comment by LingDong-** [comment]
Thanks a lot for spotting the bug! Fortunately it was an easy fix and now it should behave as expected: 57acbfe66147b2d174023c285ee5c473aa44a047

**Comment by shyoshyo** [comment]
There are still tiny bug which results in inconsistent input & output for floating numbers.

On commit `c1ec54e39a7a46d6cef122d075a4b182934acdbd`, wenyan code:
```
吾有一數。曰負一又二分三釐四毫五絲六忽七微，書之。
```
compiled to:
```JavaScript
var _ans36 = -0.765433;
console.log(_ans36);
```
which outputs:
```
負零又七分六釐五毫四絲三忽二微
```

It seems like that input `負一又二分三釐四毫五絲六忽七微` was misinterpreted as  `(負一)又(二分三釐四毫五絲六忽七微) = -1 + .234567 = -0.765433`, which is different from the format for outputting floating number.

---

BTW, on current branch `負二` `負三` ... were both misinterpreted as `-1`:

```
吾有一數。曰負三，書之。
```

```JavaScript
var _ans40 = -1;
console.log(_ans40);
```

```
負一
```

**Comment by LingDong-** [comment]
Thanks for pointing out, you're exactly right. Should be truly fixed now: 8f35872e750cf9db2100a117a1c03c16fdd6f595
I really need to get some coffee :-P

**Comment by shyoshyo** [comment]
all fixed now, thanks for your hard work!
